src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -274,7 +280,7 @@ void CslTxScheduler::HandleSentFrame(const Mac::TxFrame &aFrame, Error aError, C
         {
             aChild.SetIndirectDataSequenceNumber(aFrame.GetSequence());
```

Comment:
The sequence number may not be set as well if `IsHeaderUpdated()` is false.

Reply:
I don't think that's the case since the radio platform never takes responsibility of updating the sequence numbers. Doc for `mIsHeaderUpdated` states:

> If `mIsHeaderUpdated` is not set, then the frame counter and key id are not set in the frame by OpenThread core and it is the responsibility of the radio platform to assign them. The platform should update the frame (assign counter and key id) even if the transmission gets aborted or fails (e.g., channel access error).

Reply:
You are correct. 

But there is situation we may need to think of:
- What if the seq-num is set but the frame-counter/key-id is not (mac always sets the seq number, but radio platform may or may not update the frame-counter/key-id
- Do we need to track this situation and ensure on next retx we re-use the same seq-num but somehow mark that we don't yet have the frame-counter/key-id and need the sub-mac or radio platform to assign them.
- Need to think more on this.

Reply:
Added more details about this in https://github.com/openthread/openthread/pull/6849#issuecomment-893858383 at item no 6.